src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -81,11 +81,18 @@ protected void startupService() {
 
   @Override
   protected void shutdownService() {
-    localNoThreadScheduler.clearTasks();
+    commonSelector.wakeup();
+    for(final Client client: clients.values()) {
+      client.close();
+    }
+    for(final Server server: servers.values()) {
+      server.close();
+    }
     if(commonSelector != null && commonSelector.isOpen()) {
-      commonSelector.wakeup();
       closeSelector(schedulerPool, commonSelector);
-      commonSelector.wakeup();
+    }
+    if(localNoThreadScheduler.hasTaskReadyToRun()) {
```

Comment:
Might as well just invoke .tick(null) here.  No reason to check if it has tasks to run, since tick will return immediately if it nothing ready for immediate run (in contrast to blockingTick which would wait if necessary).